x/bonds/internal/types/bonds.go

```diff
@@ -340,9 +345,21 @@ func (bond Bond) ReserveAtSupply(supply sdk.Int) (result sdk.Dec) {
 		c := args["c"]
 		temp1 := x.Sub(b)
 		temp2 := temp1.Mul(temp1).Add(c)
-		temp3 := temp2.ApproxSqrt()
+		temp3, e := temp2.ApproxSqrt()
+		// TODO: To check
+		if e != nil {
+			panic(e)
+		}
 		temp5 := a.Mul(temp3.Add(x))
-		constant := a.Mul((b.Mul(b).Add(c)).ApproxSqrt())
+		// TODO: To check
+		// constant := a.Mul((b.Mul(b).Add(c)).ApproxSqrt())
+		approx, err := (b.Mul(b).Add(c)).ApproxSqrt()
```

Comment:
Comment can be removed as the calculations are correct:
```suggestion
		approx, err := (b.Mul(b).Add(c)).ApproxSqrt()
```